WordPress/Classes/Services/ThemeService.m

```diff
@@ -260,29 +263,110 @@ - (NSProgress *)getThemesForBlog:(Blog *)blog
 
     ThemeServiceRemote *remote = [[ThemeServiceRemote alloc] initWithWordPressComRestApi:blog.wordPressComRestApi];
     NSMutableSet *unsyncedThemes = sync ? [NSMutableSet setWithSet:blog.themes] : nil;
-    
-    NSProgress *progress = [remote getThemesForBlogId:[blog dotComID]
-                                                   page:page
-                                                success:^(NSArray<RemoteTheme *> *remoteThemes, BOOL hasMore) {
-                                                    NSArray *themes = [self themesFromRemoteThemes:remoteThemes
-                                                                                           forBlog:blog];
-                                                    if (sync) {
-                                                        [unsyncedThemes minusSet:[NSSet setWithArray:themes]];
-                                                        for (Theme *deleteTheme in unsyncedThemes) {
-                                                            if (![blog.currentThemeId isEqualToString:deleteTheme.themeId]) {
-                                                                [self.managedObjectContext deleteObject:deleteTheme];
-                                                            }
-                                                        }
-                                                    }
-                                                    
-                                                    [[ContextManager sharedInstance] saveContext:self.managedObjectContext withCompletionBlock:^{
-                                                        if (success) {
-                                                            success(themes, hasMore);
-                                                        }
-                                                    }];
-                                                } failure:failure];
-    
-    return progress;
+
+    if ([blog supports:BlogFeatureCustomThemes]) {
+        BOOL freeOnly = (blog.planID.integerValue != JetpackProfessionalYearlyPlanId
+                         && blog.planID.integerValue != JetpackProfessionalMonthlyPlanId);
```

Comment:
If we can't fix this in the API yet, I'd at least bury it inside `-[Blog supports:]`